ui/src/components/communities.tsx

```diff
@@ -50,19 +50,19 @@ export class Communities extends Component<any, CommunitiesState> {
     WebSocketService.Instance.getSite();
   }
 
-  getPageFromProps(props: any): number {
-    return props.match.params.page ? Number(props.match.params.page) : 1;
-  }
-
   componentWillUnmount() {
     this.subscription.unsubscribe();
   }
 
-  // Necessary for back button for some reason
-  componentWillReceiveProps(nextProps: any) {
-    if (nextProps.history.action == 'POP') {
-      this.state = this.emptyState;
-      this.state.page = this.getPageFromProps(nextProps);
+  static getDerivedStateFromProps(props) {
```

Comment:
I'm getting typescript warnings on all of these. 

It might also be a good idea to add `CommunitiesProps`, with `page`, in order to avoid the any type here.

Reply:
Ah whoops, first go at react-like and ts- Too used to plain js lol.  I put these as `(props: any): [Thing]Props`, Inferno doesnt seem to export the Router typings for props on static methods that I could find